src/Makefile.am

```
@@ -1248,10 +1248,44 @@ AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/include/tdep-$(arch) -I.
 AM_CCASFLAGS = $(AM_CPPFLAGS)
 noinst_HEADERS += unwind/unwind-internal.h
 
-EXTRA_DIST = $(libunwind_la_EXTRAS_ia64)
+EXTRA_DIST = $(libunwind_la_EXTRAS_ia64) abi
 
 MAINTAINERCLEANFILES = Makefile.in
 
+#
+# ABI checks
+#
+# Running the ABI checks require configuring with CFLAGS="-g -Og" and running
+# the targets explicitly with `make -C src abi-check`
+#
+# This also requires libabigail tools of at least version 2.0 to be installed.
+#
+abidw_common = --no-show-locs \
+               --no-comp-dir-path \
+               --no-corpus-path \
+               --drop-undefined-syms \
+               --drop-private-types
+
+abi-generate: $(lib_LTLIBRARIES)
+	@printf 'Generating ABI files\n'; \
+	mkdir -p abi; \
+	for lib in $(lib_LTLIBRARIES:.la=); do \
+		sofile="$$(find . -name $${lib}.so)"; \
+		abifile="abi/$${lib}-${arch}-${target_os}.abi"; \
+		abidw "$${sofile}" $(abidw_common) --out-file "$${abifile}"; \
+	done
```

honor silient rules:

```suggestion
	@mkdir -p abi; \
	for lib in $(lib_LTLIBRARIES:.la=); do \
		sofile="$$(find . -name $${lib}.so)"; \
		abifile="abi/$${lib}-${arch}-${target_os}.abi"; \
		if $(AM_V_P); then echo abidw "$${sofile}" $(abidw_common) --out-file "$${abifile}"; \
		              else echo "  GEN ABI  $${abifile}"; \
		fi; \
		abidw "$${sofile}" $(abidw_common) --out-file "$${abifile}"; \
	done
```

---

See latest revision.